core/src/main/kotlin/io/specmatic/core/pattern/StringPattern.kt

```diff
@@ -157,6 +148,11 @@ data class StringPattern (
 
     override val pattern: Any = "(string)"
     override fun toString(): String = pattern.toString()
+
+    private fun generateFromRegex(regexWithoutCaretAndDollar: String, minLength: Int, maxLength: Int?): String =
+        maxLength?.let {
+            Generex(regexWithoutCaretAndDollar).random(minLength, it)
```

Comment:
Regexes supported by Generex may also have a length, e.g. a{10}. While we are at it, do we want to handle this in case it's greater than a maxLength or less than minLength?

Reply:
We are putting the constraints to generate regex from min to max length, previously we only gave it the min length and we assumed it to be normal length, which caused it to overflow most of the time out of max range

The logic written before was a bit tricky to understand